controllers/reconcilers/configuration/configuration_reconciler.go

```diff
@@ -307,6 +325,35 @@ func (r *Reconciler) Reconcile(ctx context.Context, cr *v1.Observability, s *v1.
 				Repository:  repoUrl,
 				Source:      &configSecret,
 			}
+			// if no access token or tag present in secret attempt to create resources deployment and service
+			if repos[configSecret.Name].AccessToken == "" && repos[configSecret.Name].Tag == "" && !cr.ExternalSyncDisabled() {
+				containerResources = true
+				err = r.ReconcileResourcesDeployment(ctx, cr)
+				if err != nil {
+					metrics.IncreaseFailedConfigurationSyncsMetric()
+					log.Error(err, "failed to create configuration resources deployment")
+					return v1.ResultFailed, err
+				}
+				err = r.ReconcileResourcesService(ctx, cr)
+				if err != nil {
+					metrics.IncreaseFailedConfigurationSyncsMetric()
+					log.Error(err, "failed to create configuration resources service")
+					return v1.ResultFailed, err
+				}
+			}
+		}
+	}
+
+	// if deploying resources container ensure deployment and service are ready
+	if containerResources {
+		var status v1.ObservabilityStageStatus
+		status, err = r.waitForResourcesDeployment(ctx, cr)
+		if status != v1.ResultSuccess {
+			return status, err
+		}
+		status, err = r.waitForResourcesService(ctx, cr)
```

Comment:
do we need to wait for the service? Usually they are created nearly instantly.

Reply:
@pb82 I was encountering timing issues leading reconcile errors. I'll see if I can modify so `waitForResourcesDeployment()` is enough to avoid those errors. Thanks